### ISSUE TITLE
False-positive "unresolved reference" with <std::fs::File as std::io::Read>

### ISSUE THREAD

**Issue by chamoretto** [intellij-rust/intellij-rust#8287]
## Environment

* **IntelliJ Rust plugin version:** 0.4.163.4309-213-nightly
* **Rust toolchain version:** 1.59.0-nightly (acbe4443c 2021-12-02) x86_64-unknown-linux-gnu
* **IDE name and version:** CLion 2021.3 (CL-213.5744.254)
* **Operating system:** Windows 10 10.0
* **Macro expansion engine:** new
* **Name resolution engine:** new
* **Additional experimental features:** org.rust.cargo.features.settings.gutter, org.rust.cargo.evaluate.build.scripts, org.rust.macros.proc

Tried with both External tools: "Cargo check" and "Clippy", with option "Run external linter to analyze code on the fly" and without.

## Problem description
False-positive unresolved reference, when trait is not fully imported or used in any kind of qualified path like `<std::fs::File as std::io::Read>::` or  `<File as io::Read>::`. I'm unable to reproduce it with any other struct-trait pair, even with synthetic examples, all works well except this combination.

Successfull examples:
![image](https://user-images.githubusercontent.com/28485306/146186190-f1fa9cbf-0471-4586-ae63-386317fc5985.png)

False-positive:
![image](https://user-images.githubusercontent.com/28485306/146186255-299c552e-137a-4906-8074-7527c926c3fb.png)

Same for all other methods - `read_exact`, `read_vectored`, `read`, `read_to_end`.
All of these examples compiles fine.

## Steps to reproduce
1. Clone this repo - https://github.com/chamoretto/intellij-rust-issues
2. Checkout the branch `8287` - https://github.com/chamoretto/intellij-rust-issues/tree/8287
3. Build an example, see no errors, watch unresolved references at `main.rs` at lines 62, 71, 80, 86.

**Comment by Undin** [comment]
Fixed by #8979

**Comment by Undin** [comment]
@wyfo since your example differs from the original one, I've moved it in separate issue - #9037

BTW, `<Foo>::Bar` is actually unresolved reference. but the compiler allows you to have such references in unused path statement and consider it as a warning instead of hard error
```
warning: path statement with no effect
 --> src/lib.rs:7:5
  |
7 |     <Foo>::Bar;  // Unresolved reference: `Bar`
  |     ^^^^^^^^^^^
  |
  = note: `#[warn(path_statements)]` on by default
```

**Comment by wyfo** [comment]
> BTW, <Foo>::Bar is actually unresolved reference.

@Undin Yet this compile fine, and there is no unused:
```rust
pub enum Foo {
    Bar,
    Baz,
}
pub fn bar() -> Foo {
    <Foo>::Bar
}
```
